### PR TITLE
gh-101100: [docs] Fix Sphinx warnings in `dis` module

### PR DESCRIPTION
**Pull request by sobolevn**
Original errors:

```
./cpython/Doc/library/dis.rst:46: WARNING: py:func reference target not found: myfunc
./cpython/Doc/library/dis.rst:51: WARNING: py:func reference target not found: myfunc
./cpython/Doc/library/dis.rst:795: WARNING: py:func reference target not found: builtins.__build_class__
./cpython/Doc/library/dis.rst:852: WARNING: py:attr reference target not found: co_names
./cpython/Doc/library/dis.rst:859: WARNING: py:attr reference target not found: co_names
./cpython/Doc/library/dis.rst:898: WARNING: py:attr reference target not found: co_names
./cpython/Doc/library/dis.rst:907: WARNING: py:attr reference target not found: co_names
./cpython/Doc/library/dis.rst:1426: WARNING: c:func reference target not found: PyObject_Format
```

A couple of notes for reviewers:

1. `:func:myfunc` was rendered incorrectly (invalid signature):

   Снимок экрана 2023-03-11 в 12 02 24

   Because `:func:` is reserved for functions defined with `.. function::` directive.

2. I've silenced `!builtins.__build_class__` reference, because as far as I understand, there's no desire in documenting this feature. So, now this reference renders the same way, but just does not error out
3. `:attr:!codeobject.*` are also just ignoring warnings, because right now there's no way to address these attributes documented here: https://github.com/python/cpython/blame/5ffdaf748d98da6065158534720f1996a45a0072/Doc/reference/datamodel.rst#L956 So, we just add an ignore and slightly more context
4. `PyObject_Format` will be found after https://github.com/python/cpython/pull/102596 is merged

* Issue: gh-101100

### CODE REVIEW

**Review by CAM-Gerlach — COMMENTED** [review]
Seems otherwise good overall, but instead of disabling resolution and silencing the warnings, could ``co_names`` be properly documented instead, just like [`codeobject.co_positions`](https://docs.python.org/3.12/reference/datamodel.html#codeobject.co_positions) currently is, especially since it's referenced nearly a dozen times in this document?

**Comment by sobolevn** [comment]
I don't know enough about the history of these fields. The might be not documented on purpose. We need a field expert here to decide.

**Comment by sobolevn** [comment]
@iritkatriel can you please take a look? :)

**Review by Fidget-Spinner — APPROVED** [review]

File: Doc/library/dis.rst [hunk 1]
```diff
@@ -851,13 +851,13 @@ iterations of the loop.
...
    Implements ``name = STACK.pop()``. *namei* is the index of *name* in the attribute
-   :attr:`co_names` of the code object. The compiler tries to use
+   :attr:`!codeobject.co_names` of the code object. The compiler tries to use
    :opcode:`STORE_FAST` or :opcode:`STORE_GLOBAL` if possible.
 
 
 .. opcode:: DELETE_NAME (namei)
 
-   Implements ``del name``, where *namei* is the index into :attr:`co_names`
+   Implements ``del name``, where *namei* is the index into :attr:`!codeobject.co_names`
```
Comment (AA-Turner) [inline Doc/library/dis.rst]: I think we should prefer not prefixing with ``codeobject`` -- currently when read out loud the markup makes sense, but with this change we are saying ``namei`` is the index into the ``codeobject.co_names`` attribute of the code object, which doesn't make sense.
```suggestion
   Implements ``del name``, where *namei* is the index into :attr:`!co_names`
```

**Review by AA-Turner — CHANGES_REQUESTED** [review]
A comment on the pre-fixing of ``codeobject.`` -- I think we should revert this for now.

It may also be better to add an entry to ``nitpick_ignore`` in ``conf.py`` with an explanatory note rather than supressing these warnings, if we intend to document the code object attributes at a later date.

**Comment by hugovk** [comment]
> if we intend to document the code object attributes at a later date.

Or can we do it now?

File: Doc/library/dis.rst (hunk 1 above)
Comment (JelleZijlstra) [reply Doc/library/dis.rst]: I agree, this displays as `codeobject.co_names` which looks odd.

Comment (arhadthedev) [reply Doc/library/dis.rst]: > I think we should prefer not prefixing with codeobject

Probably, we need `~!codeobject.co_names` for this.

**Comment by sobolevn** [comment]
Closed by https://github.com/python/cpython/pull/108476

**Comment by erlend-aasland** [comment]
Sorry, @sobolevn, I didn't know you had a PR open :(

**Pull request closed without merging**